cups/http-addrlist.c

```diff
@@ -288,21 +288,36 @@ httpAddrConnect2(
       for (i = 0; i < nfds; i ++)
       {
 	DEBUG_printf("pfds[%d].revents=%x\n", i, pfds[i].revents);
-#  ifdef __sun
+
+#  ifdef _WIN32
+	if (((WSAGetLastError() == WSAEINPROGRESS) && (pfds[i].revents & POLLIN) && (pfds[i].revents & POLLOUT)) ||
+	    ((pfds[i].revents & POLLHUP) && (pfds[i].revents & (POLLIN|POLLOUT))))
+#  else
+	if (((errno == EINPROGRESS) && (pfds[i].revents & POLLIN) && (pfds[i].revents & POLLOUT)) ||
+	    ((pfds[i].revents & POLLHUP) && (pfds[i].revents & (POLLIN|POLLOUT))))
+#  endif /* _WIN32 */
+	{
+	// Initially for Solaris, but some Linux kernels generate POLLIN or POLLOUT
```

Comment:
I would drop the talk about Solaris here; "git log" and "git blame" can be used to retrieve this history, but here the point is to document why we are going through this dance.

So maybe something along the lines of:

```c
// Some systems generate POLLIN or POLLOUT together with POLLHUP when doing
// asynchronous connections. The solution seems to be to use getsockopt to
// check the SO_ERROR value and ignore the POLLHUP if there is no error or
// the error is EINPROGRESS.
```